hy/core/language.hy

```diff
@@ -369,12 +369,15 @@
   "Chunks coll into n-tuples (pairs by default). The remainder, if any, is not
    included unless a fillvalue is specified. The step defaults to n, but can be
    more to skip elements, or less for a sliding window with overlap."
-  (setv
-   step (or step n)
-   slices (genexpr (itertools.islice coll start None step) [start (range n)]))
-  (if (is fillvalue -sentinel)
-    (apply zip slices)
-    (apply zip-longest slices {"fillvalue" fillvalue})))
+  (if (< n 1) [] (do
```

Comment:
Sorry, I misread the error message. `tee` complains about negative `n`, not zero. So shall we have `partition` with `n` = 0 return infinitely many empty tuples, and raise an exception on negative `n`?